formatters/event_drop/event_drop.go

```diff
@@ -180,3 +185,15 @@ func (d *Drop) WithLogger(l *log.Logger) {
 func (d *Drop) WithTargets(tcs map[string]*types.TargetConfig) {}
 
 func (d *Drop) WithActions(act map[string]map[string]interface{}) {}
+
+func shift[T any](es []T, dropIndexes []int) []T {
+	lastDropIndexes := len(dropIndexes) - 1
+	for i, dropIndex := range dropIndexes {
+		if i == lastDropIndexes {
+			es = es[:dropIndex]
+			continue
+		}
+		copy(es[dropIndex:], es[dropIndexes[i+1]:])
+	}
+	return es
+}
```

Comment:
@karimra is the idea of this function to drop elements from `es` whos indexes match the values in `dropIndexes`?

e.g.
```go
dropIndexes := []int{0, 2, 3}
es := []string{"A", "B", "C", "D", "E", "F"}
// should return B E F ?
```

Reply:
Yes but this doesn't seem to work properly when the first message is to be dropped, will rework it.

Reply:
yep, I personally find this solution a very easy to read

https://goplay.tools/snippet/fcH6_GTddi1

it has a O(n) complexity, but it seems that the order of the events needs to be preserved, thus the re-slicing (shifting) is not considerably faster on the number of events

Reply:
My point was to not have to allocate a new event slice to append the allowed messages.